app/models/recipe.rb

```diff
@@ -2,4 +2,10 @@ class Recipe < ApplicationRecord
   belongs_to :user
   has_many :recipe_foods
   has_many :foods, through: :recipe_foods
+
+  validates :name, presence: true
+
+  def public_recipes
+    @recipes = Recipe.where(public: true)
+  end
```

Comment:
### Optional
This was not apart of this task requirement. But we need it later.